vega_sim/scenario/ideal_market_maker_v2/agents.py

```diff
@@ -245,25 +253,29 @@ def step(self, vega_state: VegaState):
             else:
                 sell_order = order
 
-        # self.vega.submit_simple_liquidity(
-        #     wallet_name=self.wallet_name,
-        #     market_id=self.market_id,
-        #     commitment_amount=self.commitment_amount,
-        #     fee=0.002,
-        #     reference_buy="PEGGED_REFERENCE_MID",
-        #     reference_sell="PEGGED_REFERENCE_MID",
-        #     delta_buy=self.bid_depth,
-        #     delta_sell=self.ask_depth,
-        # )
-
-        self._place_orders(
-            buy_offset=self.bid_depth,
-            sell_offset=self.ask_depth,
-            volume=1,
-            buy_order=buy_order,
-            sell_order=sell_order,
+        market_data = self.vega.market_data(market_id=self.market_id)
+        LOB_best_bid = market_data.best_bid_price / 10**self.mdp
+        LOB_bset_ask = market_data.best_offer_price / 10**self.mdp
+
+        self.vega.submit_simple_liquidity(
+            wallet_name=self.wallet_name,
+            market_id=self.market_id,
+            commitment_amount=self.commitment_amount,
+            fee=0.002,
+            reference_buy="PEGGED_REFERENCE_BEST_BID",
+            reference_sell="PEGGED_REFERENCE_BEST_ASK",
+            delta_buy=self.bid_depth + LOB_best_bid - self.price_process[self.current_step],
+            delta_sell=self.ask_depth + self.price_process[self.current_step] - LOB_bset_ask,
+            is_amendment=True,
         )
-        self.current_step += 1
+
+        # self._place_orders(
+        #     buy_offset=self.bid_depth,
+        #     sell_offset=self.ask_depth,
+        #     volume=1,
+        #     buy_order=buy_order,
+        #     sell_order=sell_order,
+        # )
```

Comment:
The market maker will place pegged order from liquidity provision, so I think we do not need to place limit order by hand here?

Reply:
Liquidity provision is only deployed when needed, whereas placing limit orders makes it more explicit and ensures the orders are actually on the book at all times (Having a deeper LOB means the orders may not always be deployed otherwise).